microsetta_public_api/api/tests/test_api.py

```diff
@@ -243,6 +243,118 @@ def test_metadata_sample_ids_get_null_parameters_succeeds(self):
         self.assertCountEqual(obs['sample_ids'], exp_ids)
         self.mock_method.assert_called_with()
 
+    def test_metadata_sample_ids_post_query_builder(self):
+        with self.app_context(), patch('microsetta_public_api.api.metadata'
+                                       '.filter_sample_ids_query_builder') as \
+                mock_method:
+            mock_method.return_value = jsonify({
+                'sample_ids': [
+                    'sample-1',
+                    'sample-2',
+                ],
+            })
+            _, self.client = self.build_app_test_client()
+
+        request_body = \
+            {
+                "condition": "AND",
+                "rules": [
+                    {
+                        "id": "age_years",
+                        "field": "age_years",
+                        "type": "double",
+                        "input": "number",
+                        "operator": "less",
+                        "value": 10.25
+                    },
+                ],
+                "valid": True
+            }
+
+        response = self.client.post(
+            "/api/metadata/sample_ids",
+            content_type='application/json',
+            data=json.dumps(request_body)
+        )
+        exp_ids = ['sample-1', 'sample-2']
+        self.assertStatusCode(200, response)
+        obs = json.loads(response.data)
+        self.assertCountEqual(['sample_ids'], obs.keys())
```

Comment:
count equal? Aren't the obs values known? If the ordering is not reliable, then passing the obs through `sorted` would work

Reply:
Theoretically, the ordering should be reliable. But I don't see a need for the results to have a consistent order, so make the more permissive check.

I also don't see why one method should be preferred over the other, if they make equivalent/correct checks?

Reply:
Oh, i see now. I had assumed `assertCountEqual` was testing the length of the two arguments -- my bad. I haven't seen this in use before